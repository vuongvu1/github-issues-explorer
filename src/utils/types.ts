export type IssueType = Array<{
  id: string;
  title: string;
  number: number;
  createdAt: string;
  author: {
    login: string;
  };
  comments: {
    totalCount: number;
  };
}>;
