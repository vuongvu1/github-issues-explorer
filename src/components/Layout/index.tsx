import React from "react";
import SC from "./styles";
import { Text, TextType } from "..";

interface Props {
  title?: string;
}

const Layout: React.FC<Props> = ({ title, children }) => {
  return (
    <SC.Container>
      <SC.Header>
        <Text type={TextType.H2}>{title}</Text>
      </SC.Header>
      <SC.BodyContainer>
        <SC.Body>{children}</SC.Body>
      </SC.BodyContainer>
    </SC.Container>
  );
};

export default Layout;
