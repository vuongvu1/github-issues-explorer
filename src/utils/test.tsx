import { ReactNode } from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import { MockedProvider } from "@apollo/react-testing";
import { BrowserRouter } from "react-router-dom";
import { theme } from "src/globalStyles";
import {
  IssueState,
  IssueOrderField,
  OrderDirection,
} from "src/generated/graphql";

const mockStore = configureMockStore();
const store = mockStore({
  repoSlice: {
    url: "https://github.com/reactjs/reactjs.org",
    currentPage: 1,
    searchParams: {
      name: "reactjs.org",
      owner: "reactjs",
      status: IssueState.Open,
      orderBy: IssueOrderField.CreatedAt,
      orderDir: OrderDirection.Desc,
      before: undefined,
      after: undefined,
      first: 10,
      last: undefined,
    },
  },
});

export const customRender = (Component: ReactNode) =>
  render(
    <ThemeProvider theme={theme}>
      <MockedProvider mocks={[]} addTypename={false}>
        <BrowserRouter>
          <Provider store={store}>{Component}</Provider>
        </BrowserRouter>
      </MockedProvider>
    </ThemeProvider>
  );
