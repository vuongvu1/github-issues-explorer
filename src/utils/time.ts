const ONE_YEAR_IN_SECONDS = 31536000;
const ONE_MONTH_IN_SECONDS = 2592000;
const ONE_DAY_IN_SECONDS = 86400;
const ONE_HOUR_IN_SECONDS = 3600;
const ONE_MINUTE_IN_SECONDS = 60;

export const timeSince = (date: Date) => {
  const seconds = Math.floor((new Date().getTime() - date.getTime()) / 1000);

  let interval = seconds / ONE_YEAR_IN_SECONDS;

  if (interval > 1) {
    return `${Math.floor(interval)} years`;
  }

  interval = seconds / ONE_MONTH_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} months`;
  }

  interval = seconds / ONE_DAY_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} days`;
  }

  interval = seconds / ONE_HOUR_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} hours`;
  }

  interval = seconds / ONE_MINUTE_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} minutes`;
  }

  return `${Math.floor(seconds)} seconds`;
};
