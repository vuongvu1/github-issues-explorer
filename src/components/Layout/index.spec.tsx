import { cleanup } from "@testing-library/react";
import { customRender } from "src/utils/testUtils";
import Layout from "./index";

describe("Layout", () => {
  afterEach(cleanup);

  test("should match snapshot", () => {
    const { container } = customRender(
      <Layout title="This is title">
        <p>test paragraph</p>
      </Layout>
    );

    expect(container).toMatchSnapshot();
  });
});
