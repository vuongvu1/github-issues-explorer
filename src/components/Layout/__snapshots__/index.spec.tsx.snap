// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match snapshot 1`] = `
<div>
  <div
    class="sc-bdvvaa jlIssb"
  >
    <div
      class="sc-gsDJrp iBniiG"
    >
      <h2
        class="sc-pVTma aYeDQ"
        type="h2"
      >
        This is title
      </h2>
    </div>
    <div
      class="sc-dkPtyc jwrIlo"
    >
      <div
        class="sc-hKwCoD dhgyQ"
      >
        <p>
          test paragraph
        </p>
      </div>
    </div>
  </div>
</div>
`;
